core/sdk-core/src/test/java/software/amazon/awssdk/core/internal/async/ByteArrayAsyncRequestTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.async;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.Test;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+public class ByteArrayAsyncRequestTest {
+    testSubscriber subscriber = new testSubscriber();
+    private class testSubscriber implements Subscriber<ByteBuffer> {
+        private Subscription subscription;
+        protected AtomicBoolean onCompleteCalled = new AtomicBoolean(false);
+
+        @Override
+        public void onSubscribe(Subscription s) {
+            this.subscription = s;
+            CompletableFuture<Void> completableFuture1 = CompletableFuture.runAsync(() -> s.request(1));
+            CompletableFuture<Void> completableFuture2 = CompletableFuture.runAsync(() -> s.request(1));
+            completableFuture1.join();
+            completableFuture2.join();
+        }
+
+        @Override
+        public void onNext(ByteBuffer byteBuffer) {
+            subscription.request(1);
+        }
+
+        @Override
+        public void onError(Throwable throwable) {
+
+        }
+
+        @Override
+        public void onComplete() {
+            onCompleteCalled.set(true);
+        }
+    }
+
+    @Test
+    public void byteArrayAsyncRequestTest() {
+        ByteArrayAsyncRequestBody byteArrayReq = new ByteArrayAsyncRequestBody("Hello World!".getBytes());
+        byteArrayReq.subscribe(subscriber);
+        assertThat(subscriber.onCompleteCalled);
```

Comment:
It should be `assertThat(subscriber.onCompleteCalled).isTrue()`; otherwise it's not asserting anything.